### PR TITLE
docs(adr): add internal modules ADR

### PR DESCRIPTION
**Pull request by joshblack**
Adds an ADR for managing Internal Modules in Primer React.

[Rendered Markdown](https://github.com/primer/react/blob/fb15555b1ab174a2445b128fbb8999bf54794837/contributor-docs/adrs/adr-014-internal-modules.md)

### CODE REVIEW

**Comment by joshblack** [comment]
cc @primer/react-reviewers would love to get your opinion when you have a sec! 👀 The tl;dr is that this is a convention for where to put modules within Primer React that we don't intent to include as a part of the public API of the package.

File: contributor-docs/adrs/adr-014-internal-modules.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    "./feature/internal/*": null
+  }
+}
+```
+
+
+### Impact
+
+- Update the `"exports"` field in `package.json` to exclude `./lib-esm/internal` from usage
+- (In a major release) Move internal-only files into internal folder
```
Comment (broccolinisoup) [inline contributor-docs/adrs/adr-014-internal-modules.md:61]: How would that affect the usages of these internal files? It makes total sense to do in a major release but I am wondering would we provide any alternatives or make exceptions to exclude some files if they are in used but intended to be internal? I hope it makes sense!

Comment (joshblack) [reply contributor-docs/adrs/adr-014-internal-modules.md:61]: Hey @broccolinisoup! 👋 

Great point, I'm glad you brought it up. If something is meant to be internal but it has usage in dotcom then we should definitely continue to export it from the bundle, in my opinion. I think at that point it stops being internal and is something that teams have come to rely on (albeit accidentally) and clearly has value for them and for us to support.

Curious what you think 👀

Comment (broccolinisoup) [reply contributor-docs/adrs/adr-014-internal-modules.md:61]: Yeah, I agree!

**Review by colebemis — APPROVED** [review]
Seems reasonable 👍

**Pull request merged**